src/coreclr/src/vm/bundle.cpp

```diff
@@ -59,17 +60,18 @@ BundleFileLocation Bundle::Probe(LPCWSTR path, bool pathIsBundleRelative) const
     //    Bundle.Probe("path/to/exe/lib.dll") => m_probe("lib.dll")
     //    Bundle.Probe("path/to/exe/and/some/more/lib.dll") => m_probe("and/some/more/lib.dll")
 
+    StackScratchBuffer scratchBuffer;
+    LPCSTR utf8Path(path.GetUTF8(scratchBuffer));
+
     if (!pathIsBundleRelative)
     {
-        size_t baseLen = m_basePath.GetCount();
-
 #ifdef TARGET_UNIX
-        if (wcsncmp(m_basePath, path, baseLen) == 0)
+        if (wcsncmp(m_basePath, path, m_basePath.GetCount()) == 0)
```

Comment:
We are not doing any additional precomputation, the ctor has the UTF8 length of m_basePath for Set call, we are caching and using it here. If I am reading it right, this `+= cached_int32` solution is comparatively much less complicated than what ::Delete does (calculates length, insert and copy).

Reply:
Have you verified that the += really works as expected? It seems strange that it would, please see my other comment on it.

Reply:
Yes, it works because line74 is `LPCSTR+=int` and not `SString+=int`. When it wasn't working in previous run, we had these CI failures on all installer legs (modolo osx-arm64):
<img width="626" alt="image" src="https://user-images.githubusercontent.com/3840695/98885183-42403600-249a-11eb-89a4-a040f5002f74.png">

Reply:
Ah, I was looking at the version where there still was SString at line 74 when writing that comment. I am sorry, I got confused by the evolution of this change.